### PR TITLE
Chess Sound Addition

### PR DESCRIPTION
**Pull request by Fouad-sys**

### CODE REVIEW

File: mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/application/SpeechFacadeTest.kt [hunk 1]
```diff
@@ -31,23 +50,48 @@ class SpeechFacadeTest {
...
+    val soundPlayer = AndroidSoundPlayer(context = context, mediaPlayer = mediaPlayer)
+    val facade =
+        SpeechFacade(
+            UnknownCommandSpeechRecognizerFactory,
+            FakeTextToSpeechFactory,
+            soundPlayer,
+            emptyDataStoreFactory())
+    facade.synthesize("Pawnies")
+    assertThat(called).isTrue()
+  }
```
Comment (alexandrepiveteau) [inline mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/application/SpeechFacadeTest.kt:96]: This should not be in `SpeechFacadeTest`. It looks like you're testing both the `AndroidSoundPlayer` and the `SpeechFacade` simultaneously.

**Review by alexandrepiveteau — CHANGES_REQUESTED** [review]
Looking good, but some tests are not in the right place or test two things at once.

File: mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/Dsl.kt [hunk 2]
```diff
@@ -262,6 +270,7 @@ fun ComposeContentTestRule.setContentWithTestEnvironment(
     auth: Auth = emptyAuth(),
     assets: AssetManager = emptyAssets(),
     recognizer: SpeechRecognizerFactory = FailingSpeechRecognizerFactory,
+    synthesizer: TextToSpeechFactory = FakeTextToSpeechFactory,
```
Comment (alexandrepiveteau) [inline mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/Dsl.kt:273]: Missing doc

Comment (alexandrepiveteau) [reply mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/Dsl.kt:273]: Also, you should provide the `SoundPlayer` as an argument.

**Review by alexandrepiveteau — DISMISSED** [review]
LGTM

**Review by matt989253 — APPROVED** [review]
Nice feature! And love the sound
(But there seems to be a failing test)

**Review by alexandrepiveteau — APPROVED** [review]

**Pull request merged**